### ISSUE TITLE
Autosave updating fields after DB submit (#978)

### ISSUE THREAD

**Issue by a1pilot** [nextcloud/contacts#1186]
Contacts 3.1.3

Re-installed to check bug fixes.

#978 is back in the latest version.

**Comment by skjnldsv** [comment]
Please provide more informations.
Which field did you edited? Did it rolledback to a previous state while you were typing? Does it do that with all fields?

Thanks

**Comment by a1pilot** [comment]
It's exactly the same bug as #978. Whatever happened there appears to be back. I get halfway through a field and autosave kicks in and wipes out what I've typed.

> Describe the bug

> When creating a new contact in the app on Nextcloud, it autosaves very frequently as you type. As I start typing a name, say "Alice Appleby", it autosaves part way through and loses the last few characters.

> What this looks like is I type "Alice Appleby", but it will autosave halfway at "Alice App" and then reset the field to "Alice App", losing everything I typed after that point.

> It's worse if there's more text, e.g. "Alice Appleby Smith", because the moment it autosaves and resets it goes back to "Alice App" and then continues typing, so you end up with "Alice Appmith".

> I can cope with it on the occasional new contact, but to try and add several it is really frustrating. I've lost several phone numbers because I didn't see this and now I have the start and end only.

> To Reproduce
> Steps to reproduce the behavior:
> As above. Try to create a new contact and as you type monitor for autosave jumping around the text fields.

> Expected behavior
> Contact should be created, with autosave being a passive event - it should autosave whatever the contents are at the time of actioning, but not feedback to the fields.

> Actual behavior
> Autosave is submitting to the database and then updating the fields on the page with whatever was just submitted. Wiping out anything not submitted but accepted into the fields, post-save event.

**Comment by skjnldsv** [comment]
Thanks!
let me try again.

Are you doing this on a slow server? Slow machine? Mobile?
I will also need a screenshot of your network requests after this happened.
Before you do anything, still on the development tools, there is a tab called `network`. Click on it and then the xhr filter. Do your action and screenshot the network log like this:
![capture d ecran_2018-09-27_21-49-46](https://user-images.githubusercontent.com/14975046/46170857-43548000-c29f-11e8-9403-db72713dd9c1.png)

**Comment by a1pilot** [comment]
Gigabit network. Xeon X3 quad, 24 GB RAM, 128 GB SSD, Debian 10. Ping is a bit higher than I'd like.

This is the network results for the New Contact process:
![image](https://user-images.githubusercontent.com/665111/61696024-71261c00-ad2c-11e9-83cb-fdcbb3d0b166.png)

I can now see it only happens on the first autosave of a new contact. Subsequent saves don't experience the issue.

So: new contact, enter phone number "01325", quick pause to trigger autosave, continue entering "456378", results in the loss of "456378". But, if you then update the phone number after the first autosave it just works as it should.

**Comment by skjnldsv** [comment]
Ah nice catch!
Thanks for that!

I indeed see a 201 request so it only do that for newly created contacts.

**Comment by jangop** [comment]
I face the same issue, on every new contact I create.

**Comment by skjnldsv** [comment]
Fixed already
